src/main/java/com/mraof/minestuck/inventory/AnthvilMenu.java

```diff
@@ -0,0 +1,111 @@
+package com.mraof.minestuck.inventory;
+
+import com.mraof.minestuck.block.MSBlocks;
+import com.mraof.minestuck.inventory.slot.InputSlot;
+import com.mraof.minestuck.item.MSItems;
+import net.minecraft.core.BlockPos;
+import net.minecraft.network.FriendlyByteBuf;
+import net.minecraft.world.entity.player.Inventory;
+import net.minecraft.world.entity.player.Player;
+import net.minecraft.world.inventory.*;
+import net.minecraft.world.item.ItemStack;
+import net.minecraftforge.items.IItemHandler;
+import net.minecraftforge.items.ItemStackHandler;
+import net.minecraftforge.items.SlotItemHandler;
+
+import javax.annotation.Nonnull;
+
+public class AnthvilMenu extends AbstractContainerMenu
+{
+	private static final int uraniumInputX = 143;
+	private static final int uraniumInputY = 54;
+	private static final int itemInputX = 81;
+	private static final int itemInputY = 35;
+	
+	private final DataSlot fuelHolder;
+	private final ContainerLevelAccess levelAccess;
+	
+	public AnthvilMenu(int windowId, Inventory playerInventory, FriendlyByteBuf buffer)
```

Comment:
The machine position is no longer needed for this menu and can be cleared out. The buffer argument can then be cleared out along with it (meaning that you'd skip the `IContainerFactory` cast in `MSMenuTypes` and the position argument when calling `NetworkHooks.openScreen()` from the block class.